ui/src/lib/preprocessItinerary.ts

```diff
@@ -41,12 +47,13 @@ export const preprocessItinerary = (from: Location, to: Location) => {
 		joinInterlinedLegs(it);
 	};
 
-	return (r: { data: PlanResponse | undefined; error: unknown }) => {
+	return (r: Awaited<RequestResult<PlanResponse, ApiError, false>>): PlanResponse => {
+		if (r.error?.error) throw new Error(r.error.error);
 		if (r.error) throw new Error(String(r.error));
 
-		r.data!.itineraries.forEach(updateItinerary);
-		r.data!.direct.forEach(updateItinerary);
+		r.data.itineraries.forEach(updateItinerary);
```

Comment:
Due to the better typing of `r`, the non-null assertions can be removed. I noticed in this case, but in general it is something typescript-eslint can catch with type-aware linting, which might be useful to have, but I don't know how to set that up in .svelte files.

https://typescript-eslint.io/rules/no-unnecessary-type-assertion/